apps/shared/avifpng.c

```diff
@@ -448,6 +486,41 @@ avifBool avifPNGWrite(const char * outputFilename, const avifImage * avif, uint3
     if (avif->icc.data && (avif->icc.size > 0)) {
         png_set_iCCP(png, info, "libavif", 0, (png_iccp_datap)avif->icc.data, (png_uint_32)avif->icc.size);
     }
+
+    const int numTextMetadataChunks = (avif->exif.data && (avif->exif.size > 0)) + (avif->xmp.data && (avif->xmp.size > 0));
+    if (numTextMetadataChunks != 0) {
+        png_text texts[2];
+        if (avif->exif.data && (avif->exif.size > 0)) {
+            if (!avifBytesToRawProfile(&avif->exif, "Exif", &exifRawProfile)) {
+                fprintf(stderr, "Error writing PNG: Exif metadata is too big\n");
+                goto cleanup;
+            }
+            const png_text text = {
+                .compression = PNG_TEXT_COMPRESSION_zTXt,
+                .key = "Raw profile type exif",
+                .text = (char *)exifRawProfile.data,
+                .text_length = exifRawProfile.size,
+            };
+            texts[0] = text;
+            // Note: png_set_eXIf_1() could be used but eXIf is likely less compatible than
+            //       the raw profile zTXt method.
+        }
+        if (avif->xmp.data && (avif->xmp.size > 0)) {
+            if (!avifBytesToRawProfile(&avif->xmp, "XMP", &xmpRawProfile)) {
+                fprintf(stderr, "Error writing PNG: XMP metadata is too big\n");
+                goto cleanup;
+            }
+            const png_text text = {
+                .compression = PNG_TEXT_COMPRESSION_zTXt,
+                .key = "Raw profile type xmp",
```

Comment:
Same for XMP: the only standardized way of storing it is in the uncompressed (and not coded) iTXt chunk w/ the "XML:com.adobe.xmp" key (see XMP spec part 3, section 1.1.5). Let's drop the raw profiles if possible.